### PR TITLE
Plan of Nest's getting started

### PR DESCRIPTION
**Pull request by Pedrochu**
This is the final plan for the second version of the Getting Started guide for Nest/Finest. The first couple of chapters will arrive once this PR is approved.

### CODE REVIEW

File: Nest/00. Plan.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+### 2.3.1 Installing a package
+### 2.3.2 Uninstalling a package
+### 2.3.3 Troubleshooting
+
+## 2.4 Updating your system
+### 2.4.1 Troubleshooting
+
+## 2.5 Conclusion
+
+# 3 Some more advanced concepts
```
Comment (Arignir) [inline Nest/00. Plan.md]: I think a simpler sentence like `Advanced concepts` would be better, (less words with the same meaning) what do you think?

File: Nest/00. Plan.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+## 3.1 More about the repositories
```
Comment (Arignir) [inline Nest/00. Plan.md]: Same here, simply `Repositories` or `Managing repositories` makes it more readable

File: Nest/00. Plan.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+### 3.1.1 The official repositories
+### 3.1.2 The unofficial repositories
+### 3.1.3 Changing the repositories' configuration file
```
Comment (Arignir) [inline Nest/00. Plan.md]: Changing is meh, looks a bit like frenglish even though it's correct English.

Maybe `Editting repositories' configuration`?

File: Nest/00. Plan.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+## 4.3 Finest's commands
+### 4.3.1 Installing and removing packages through requirements
+#### 4.3.1.1 Adding a requirement
+#### 4.3.3.2 Removing a requirement
+#### 4.3.3.3 Troubleshooting
+
+### 4.3.2 Updating your system by merging your dependency graph
+#### 4.3.2.1 Troubleshooting
+
+### 4.3.3 Actions on groups
```
Comment (Arignir) [inline Nest/00. Plan.md]: `Groups` or `Group management` seems more straightforward to me. What do you think?

File: Nest/00. Plan.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+## 4.1 Key concepts
+### 4.1.1 The cache
+### 4.1.2 The transactions
+### 4.1.3 The dependency graph
+### 4.1.4 Groups
+### 4.1.5 Example
+
+## 4.2 Finest's options
+
+## 4.3 Finest's commands
```
Comment (Arignir) [inline Nest/00. Plan.md]: 4 level of titles will probably not look good on the title bar when [guides](https://github.com/raven-os/guides-raven-os-org) will be finished. I'd like to keep it to three levels as much as possible. Therefore, I think we can inline this one, it doesn't add anything big, at least not form my PoV. What do you think?

File: Nest/00. Plan.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+### 4.3.1 Installing and removing packages through requirements
```
Comment (Arignir) [inline Nest/00. Plan.md]: This title is quite long, can we shorter it? If not, that's fine.

File: Nest/00. Plan.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+#### 4.3.1.1 Adding a requirement
+#### 4.3.3.2 Removing a requirement
+#### 4.3.3.3 Troubleshooting
+
+### 4.3.2 Updating your system by merging your dependency graph
```
Comment (Arignir) [inline Nest/00. Plan.md]: Same here.

File: Nest/00. Plan.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+## 2.2 Nest commands
+### 2.2.1 Synchronizing the local cache
+### 2.2.2 Troubleshooting
+
+## 2.3 Installing and uninstalling a package
+### 2.3.1 Installing a package
+### 2.3.2 Uninstalling a package
+### 2.3.3 Troubleshooting
+
+## 2.4 Updating your system
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: System may be unclear or vague."operating system" or even "OS" (with a description when then abbreviation is introduced) would be more precise.

File: Nest/00. Plan.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+### 2.4.1 Troubleshooting
+
+## 2.5 Conclusion
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: I have no Idea what this means and what this section would contain. Probably a bad sign. I imagine maybe this should "summary" or something along those lines?

File: Nest/00. Plan.md (hunk 7 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: I'd be against the use of *your*. Something like "Updating the system by merging its dependency graph" would sound more professional (the current tone could even sound condescendent) and (IMO) clearer.

File: Nest/00. Plan.md (hunk 4 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: I could also imagine "Modifying groups".

File: Nest/00. Plan.md (hunk 8 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: (also the "your" could be replaced by "the")

File: Nest/00. Plan.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+#### 4.3.1.1 Adding a requirement
+#### 4.3.3.2 Removing a requirement
+#### 4.3.3.3 Troubleshooting
+
+### 4.3.2 Updating your system by merging your dependency graph
+#### 4.3.2.1 Troubleshooting
+
+### 4.3.3 Actions on groups
+#### 4.3.3.1 Creating a group
+#### 4.3.3.2 Removing a group
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: I think "Deleting a group" would better convey the meaning I imagine you're looking for.

File: Nest/00. Plan.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
+# 1 Introduction
+## 1.1 Package manager
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: "What is a package manager" would be a more common title. This depends on the tone of the document. This could however help avoid the repeated titles (and potential confusions).
This avoids for example the awkward "More about repositories".

File: Nest/00. Plan.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+## 1.3 Repositories
+
+## 1.4 Package's syntax
+
+# 2 Nest
+## 2.1 Nest's options
+
+## 2.2 Nest commands
+### 2.2.1 Synchronizing the local cache
+### 2.2.2 Troubleshooting
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: Troubleshooting what? I know you want me to know from context, I just feel you should be precise.

Comment (Gaspard--) [reply Nest/00. Plan.md]: (this applies to other sections with the same name)

File: Nest/00. Plan.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+### 2.3.3 Troubleshooting
+
+## 2.4 Updating your system
+### 2.4.1 Troubleshooting
+
+## 2.5 Conclusion
+
+# 3 Some more advanced concepts
+## 3.1 More about the repositories
+### 3.1.1 The official repositories
```
Comment (Gaspard--) [inline Nest/00. Plan.md:24]: I would drop the "The". You could eventually keep it if you are really listing of which repositories are official and unofficial, I'd explain the concepts in the section 3.1 in that case (you were maybe already planning to do this).

File: Nest/00. Plan.md (hunk 3 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: What about "Configuring repositories"?

File: Nest/00. Plan.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+### 3.1.2 The unofficial repositories
+### 3.1.3 Changing the repositories' configuration file
+
+## 3.2 The cache
+
+## 3.3 Version requirement
+
+# 4 Finest
+## 4.1 Key concepts
+### 4.1.1 The cache
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: Difference with 3.2 ? You could change 3.2 to "Overview of the cache" and then this to "The cache in detail" or something along those lines.

File: Nest/00. Plan.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+### 4.1.2 The transactions
+### 4.1.3 The dependency graph
+### 4.1.4 Groups
+### 4.1.5 Example
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: Example of what? (I'd be more precise)

**Review by Gaspard-- — COMMENTED** [review]
It's late, I'm sleepy and I don't know why I reviewed this, so I may be completely of the mark.
Some comments are for a specific case but apply to other similar situations. Contrary to @Arignir I prefer long and precise titles when reading documents. Short titles tend to be too vague from my point of view. That doesn't mean useless words shouldn't be trimmed of course.

File: Nest/00. Plan.md (hunk 7 above)
Comment (Arignir) [reply Nest/00. Plan.md]: That's a good point

File: Nest/00. Plan.md (hunk 5 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: yeah I can just keep the third level and not detail any further in the plan

File: Nest/00. Plan.md (hunk 6 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Managing packages through requirements`?

File: Nest/00. Plan.md (hunk 7 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Updating the system with the dependency graph`?
or simply `Merging the dependency graphs`, but it almost sounds too vague to me so idk

File: Nest/00. Plan.md (hunk 11 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: In this case I can also have a question for the title of 1.2, like `What are dependencies?`

File: Nest/00. Plan.md (hunk 13 above)
Comment (Pedrochu) [reply Nest/00. Plan.md:24]: That's what I was planning to do c:

File: Nest/00. Plan.md (hunk 10 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Deleting` feels better, but the command is something like `finest remove`, so I felt like using the same verb was more relevant. I don't know if I should keep it like that or not though

File: Nest/00. Plan.md (hunk 12 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Common issues`?

File: Nest/00. Plan.md (hunk 4 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Group management` sounds better imo
Edit: resolving this as we seem to agree on `Managing groups`

File: Nest/00. Plan.md (hunk 15 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Example of transactions with the dependency graph`?
This part is basically going to be a sort of use case, where packages get installed and removed, and you would see how the dependency graph looks like after each transaction

File: Nest/00. Plan.md (hunk 10 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: @Arignir Maybe finest needs to change in that case :^)

File: Nest/00. Plan.md (hunk 12 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: "Common issues when using Nest" or "Troubleshooting nest". I didn't mean changing the word itself rather being clear what this is about.

File: Nest/00. Plan.md (hunk 10 above)
Comment (Arignir) [reply Nest/00. Plan.md]: The commands actually are `finest group add` and `finest group remove` (and not `finest group create` and `finest group remove` like the titles would suggest).

Surely that's something we can change, but the reason why they are name that way is to create a parallel with the requirements subcommands (`finest requirement add` and `finest requirement remove`)

What do @doom thinks of this?

File: Nest/00. Plan.md (hunk 12 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Troubleshooting Nest` wouldn't work because there is troubleshooting sections for almost every command, so the title would need to be more specific, like `Troubleshooting with the synchronization of the local cache with Nest` or something like that, and that is way too long and pretty redundant with the title of the part just before
So I think sticking with simply `Troubleshooting` or `Common issues` is better for those sections

File: Nest/00. Plan.md (hunk 10 above)
Comment (doom) [reply Nest/00. Plan.md]: Obviously, the titles should match the subcommands.

Then, as to how these subcommands should be named, `remove` was picked to maintain homogeneousness with the other commands (`nest remove` and `finest requirement remove`), and because `removing a group` is actually `removing (a requirement to) the packages in this group`.
In the same way, `adding a group` is `creating a group and adding a requirement to it`, and not only `creating the group`.

The names / behaviors are of course open to debate if anyone wants to :smiley:

File: Nest/00. Plan.md [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+### 4.1.5 Example of transactions with the dependency graph
+
+## 4.2 Finest's options
+
+## 4.3 Finest's commands
+### 4.3.1 Managing packages through requirements
+
+### 4.3.2 Updating the OS by merging the dependency graph
+
+### 4.3.3 Group management
```
Comment (doom) [inline Nest/00. Plan.md]: I'd rather put "Managing groups" here, since the other items in this section all use verbs

File: Nest/00. Plan.md [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+
+## 1.3 Repositories
+
+## 1.4 Package's syntax
+
+# 2 Nest
+## 2.1 Nest's options
+
+## 2.2 Nest commands
+### 2.2.1 Synchronizing the local cache
```
Comment (doom) [inline Nest/00. Plan.md]: Maybe this title could be made a bit simpler ?
IMHO, an end user would rather think about "Updating the list of available packages" rather than "Synchronizing the local cache", so that would make this item easier to find for him/her.

File: Nest/00. Plan.md (hunk 8 above)
Comment (Arignir) [reply Nest/00. Plan.md]: I actually disagree with you on this. I think the system is the least-worst word to describe the whole thing.

I fear "Updating the Operating System" has too much of a low level connotation and may be mixed up with the idea of "Updating the kernel", which may not reflect the idea of updating user software (with GUI).

File: Nest/00. Plan.md (hunk 10 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: I don't know, I was shocked that you were opposing create and remove, but if it's add and remove it seems ok for me.

File: Nest/00. Plan.md (hunk 12 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: Well I think you already know my position: I'd prefer a long and precise title

File: Nest/00. Plan.md (hunk 8 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: Agreed about the confusion concerns. Better to keep as is.

File: Nest/00. Plan.md (hunk 12 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: It would be something like "Common issues when synchronizing the local cache using Nest".

File: Nest/00. Plan.md (hunk 15 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: What about "Dependency graph examples" ? I think the transactions themselves aren't the focus themselves from what I'm gathering ?

File: Nest/00. Plan.md (hunk 16 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: Me too :P (https://github.com/raven-os/getting-started/pull/1#discussion_r279592496)

File: Nest/00. Plan.md (hunk 17 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: To combat this sort of problem, you could have a quick "also know as" section below titles, although tbh I've never seen that in other user guides / specifications.

File: Nest/00. Plan.md (hunk 8 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: :+1: I'll put it back as 'system'

File: Nest/00. Plan.md (hunk 10 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: hmm I think I will simply change it as 'Creating a group' and 'Deleting a group'.
If I use 'add' and 'remove', I feel like `adding a group` sounds a bit weird (when `removing a group` doesn't), and `creating a group and adding a requirement to it` is a too long title :(

File: Nest/00. Plan.md (hunk 12 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: a long title like this one is pretty redundant with the title of the part just before `Synchronizing the local cache` I don't like having two titles that are so similar

File: Nest/00. Plan.md (hunk 17 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: Maybe I could write in the chapter that it's about synchronizing the local cache, but not in the plan then

File: Nest/00. Plan.md (hunk 15 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Dependency graph examples` makes me think that there will be literally examples of static dependency graphs and that's it
I think the idea of transactions in the title is still important, like maybe `Example of how transactions are altering the dependency graph`, even though it's a tiny bit long, it's pretty accurate

File: Nest/00. Plan.md (hunk 12 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: Maybe I can simply not detail these `Common issues` parts in the plan, but put them in each command's chapter

File: Nest/00. Plan.md (hunk 10 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: I think it's important to respect the command names.

File: Nest/00. Plan.md (hunk 12 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: Yes, that would be more logical actually.

File: Nest/00. Plan.md (hunk 15 above)
Comment (Gaspard--) [reply Nest/00. Plan.md]: "Dependency graph transaction examples" could potentially work to.
Also for your suggestion you can get rid of most of the verbs:
"Example of transactions altering the dependency graph"

Comment (Pedrochu) [reply Nest/00. Plan.md]: I prefer the first title you proposed. But I would rephrase it as `Example: dependency graph transactions` or `Example of dependency graph transactions`, in order to really emphasize on the word `Example`

File: Nest/00. Plan.md (hunk 7 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: What about `Updating the system from the dependency graph`?

File: Nest/00. Plan.md (hunk 10 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: Well I just realized that we decided to keep 3 levels of titles in the plan, and the ones we are arguing on are 4th level titles, so I think we don't care about those for the plan anymore :p

File: Nest/00. Plan.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
+# 1 Introduction
+## 1.1 What is a package manager?
+
+## 1.2 Packages and dependencies
+
+## 1.3 Repositories
+
+## 1.4 Package's syntax
```
Comment (Gaspard--) [inline Nest/00. Plan.md]: I would prefer
```suggestion
## 1.4 Package name syntax
```

File: Nest/00. Plan.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+### 4.1.3 The dependency graph
+### 4.1.4 Groups
+### 4.1.5 Example of dependency graph transactions
+
+## 4.2 Finest's options
+
+## 4.3 Finest's commands
+### 4.3.1 Managing packages through requirements
+
+### 4.3.2 Updating the system from the dependency graph
```
Comment (Gaspard--) [inline Nest/00. Plan.md]:
```suggestion
### 4.3.2 Updating the system through the dependency graph
```
Unless I'm understanding the intent wrong. "from" sounds weird.

File: Nest/00. Plan.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+## 1.1 What is a package manager?
+
+## 1.2 Packages and dependencies
+
+## 1.3 Repositories
+
+## 1.4 Package's syntax
+
+# 2 Nest
+## 2.1 Nest's options
```
Comment (Gaspard--) [inline Nest/00. Plan.md]:
```suggestion
## 2.1 Nest options
```
I think the apostrophe while grammatically correct should be omited.

File: Nest/00. Plan.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+
+# 4 Finest
+## 4.1 Key concepts
+### 4.1.1 The cache in detail
+### 4.1.2 The transactions
+### 4.1.3 The dependency graph
+### 4.1.4 Groups
+### 4.1.5 Example of dependency graph transactions
+
+## 4.2 Finest's options
```
Comment (Gaspard--) [inline Nest/00. Plan.md]:
```suggestion
## 4.2 Finest options
```
(same reason as above)

File: Nest/00. Plan.md [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+## 4.3 Finest's commands
```
Comment (Gaspard--) [inline Nest/00. Plan.md]:
```suggestion
## 4.3 Finest commands
```
(Same reason as above)

File: Nest/00. Plan.md [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+### 3.1.3 Configuring repositories
+
+## 3.2 Overview of the cache
+
+## 3.3 Version requirement
+
+# 4 Finest
+## 4.1 Key concepts
+### 4.1.1 The cache in detail
+### 4.1.2 The transactions
```
Comment (Gaspard--) [inline Nest/00. Plan.md]:
```suggestion
### 4.1.2 Transactions
```
I don't think there's a specific set of transactions right?

**Review by Gaspard-- — COMMENTED** [review]
I think it's important to move on at some point, nevertheless, here are a few suggestions

File: Nest/00. Plan.md (hunk 18 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: yeah, this makes more sense

File: Nest/00. Plan.md (hunk 19 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: When you update the system with Finest, you're merging a copy of your dependency graph (with all the transactions you performed) with the graph the system is currently using. So I guess in a way, both are correct :thinking: 
Although I think that `through` sounds better

File: Nest/00. Plan.md (hunk 7 above)
Comment (Arignir) [reply Nest/00. Plan.md]: The original was fine, no? It contains the idea of merging, which is the command used, so why change it?

File: Nest/00. Plan.md (hunk 23 above)
Comment (Arignir) [reply Nest/00. Plan.md]: (Actually there is (`PULL`, `INSTALL`,`UNINSTALL` and `UPDATE`), but `Transactions` is still a better name :D)

File: Nest/00. Plan.md (hunk 7 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: I was trying to shorten the title D:
but yeah I agree, the idea of merging is important for the title. `Updating the system by merging the dependency graph` it is then

File: Nest/00. Plan.md (hunk 23 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: `Transactions` sounds better indeed. `The transactions` seem like I'm going to list them all in this section (the point of this section is rather to explain the concept of transactions than listing them)

File: Nest/00. Plan.md (hunk 19 above)
Comment (Pedrochu) [reply Nest/00. Plan.md]: So from those comments [there](https://github.com/raven-os/getting-started/pull/1#discussion_r280994531), the title of this part will be `Updating the system by merging the dependency graph`, which solves that problem since there's no `from` nor `through` :p

**Review by Gaspard-- — APPROVED** [review]
LGTM

**Review by Arignir — APPROVED** [review]
